### PR TITLE
CI: update pnpm action

### PR DESCRIPTION
**Pull request by RobinKnipe**
Bump the pnpm github action, and the pnpm version.

### CODE REVIEW

File: .github/workflows/deploy-docs.yml [hunk 1]
```diff
@@ -42,13 +42,11 @@ jobs:
         run: |
           cd apps/docs/
           npm run package:serverless
-          mkdir -p .netlify
+          mkdir -p .netlify/functions publish
```
Comment (daniel-ac-martin) [inline .github/workflows/deploy-docs.yml]: `.netlify/publish` by the looks of it.

Comment (RobinKnipe) [reply .github/workflows/deploy-docs.yml]: Good spot - fixed!

**Comment by daniel-ac-martin** [comment]
LGTM, although we should raise a similar one on: https://github.com/daniel-ac-martin/NotGovUK

Do you think there will be any downsides to the upgrade? What's triggered this?

**Comment by RobinKnipe** [comment]
> LGTM, although we should raise a similar one on: https://github.com/daniel-ac-martin/NotGovUK

I'll raise that next...

> Do you think there will be any downsides to the upgrade? What's triggered this?

It's just bit-rot. Looks like it has been resolved by newer dependencies in the new version.
I looked at the motivation for the action's major version increase, it looks like a couple of props have been discontinued: 

> - Use https://pnpm.js.org/pnpm.js to install pnpm.
> - Removed inputs.bin_dest.
> - Removed inputs.registry.

As far as pnpm goes - this isn't the most recent version (I don't think), but - it's a version I've been successfully using locally for a while.

**Pull request merged**